### PR TITLE
Sort and map the emails for better reporting

### PR DESCRIPTION
**Pull request by droidpl**
@steve-norman-rft As we discussed this PR adds the sorting and mapping of users to solve some of the issues commented. It also fixes some new lint rules that were causing some errors

Feel free to test the solution out using the hash of the commit cf119da1671a09ce9720fdb798e7b9ce52154770 so we can verify all is working as expected. If so, I can make another release with these changes

### CODE REVIEW

File: src/reporter.ts [hunk 1]
```diff
@@ -110,12 +115,18 @@ function getCSVFormat(
       type: Membership.PENDING_INVITE
     }
   })
-  const allMembers = members.concat(outsideCollaborators).concat(pendingInviteMembers)
+
+  const allMembers = members
+    .concat(outsideCollaborators)
+    .concat(pendingInviteMembers)
+    .sort((a, b) => a.login.localeCompare(b.login))
```
Comment (stoe) [inline src/reporter.ts:122]: 🐙🐱💖

File: src/reporter.ts [hunk 2]
```diff
@@ -110,12 +115,18 @@ function getCSVFormat(
...
+
   const membersContent = [
     ['Login', 'Emails', 'Orgs', 'Membership', 'Created At'],
     ...allMembers.map(item => [
       item.login,
-      item.emails.join(','),
+      // Make sure the email only appears once
+      Array.from(new Set(item.emails)).join(','),
```
Comment (stoe) [inline src/reporter.ts:129]:
```suggestion
      [...new Set(item.emails)].join(','),
```

Might be the easier syntax.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#remove_duplicate_elements_from_the_array

File: src/reporter.ts [hunk 3]
```diff
@@ -43,13 +43,16 @@ function getMarkdownFormat(
...
-          item.orgs.join(','),
-          item.type.toString(),
-          item.createdAt
-        ])
+        ...allMembers
+          .sort((a, b) => a.login.localeCompare(b.login))
+          .map(item => [
+            item.login,
+            // Make sure the email only appears once
+            Array.from(new Set(item.emails)).join(','),
```
Comment (stoe) [inline src/reporter.ts:51]:
```suggestion
            [...new Set(item.emails)].join(','),
```

Might be the easier syntax.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#remove_duplicate_elements_from_the_array

**Review by stoe — COMMENTED** [review]
@droidpl, you'll probably want to run a `npm run build` and add the changes to `dist/` before we merge this?

**Comment by steve-norman-rft** [comment]
Failed when ran with the latest sha (which won't have the dist updated but I don't think it is that causing it)

```
2021-08-25T16:21:38.1134879Z ##[group]Run ActionsDesk/enterprise-members-report-action@8022e44a8252d8466869f2bf9f927594e04f248d
2021-08-25T16:21:38.1135912Z with:
2021-08-25T16:21:38.1138207Z   token: ***
2021-08-25T16:21:38.1139272Z   enterprise: ***
2021-08-25T16:21:38.1139757Z   format: html
2021-08-25T16:21:38.1140147Z ##[endgroup]
2021-08-25T16:21:48.7006460Z ##[error]Cannot read property 'trim' of undefined
Please report this to https://github.com/markedjs/marked.
2021-08-25T16:21:48.7270438Z Cleaning up orphan processes
```

**Comment by droidpl** [comment]
@stoe you're right. I will review also the issue that we are experiencing and see if it's caused by any change here or something else as it looks fishy

**Pull request merged**